components/brave_wallet/browser/bitcoin/bitcoin_wallet_service.cc

```diff
@@ -594,13 +653,15 @@ void CreateTransactionTask::OnFetchRawTransactions(
 
 double CreateTransactionTask::GetFeeRate() {
   DCHECK(!estimates_.empty());
+  double rate = 0.0;
   if (estimates_.contains(kMediumPriorityTargetBlock)) {
-    return estimates_.at(kMediumPriorityTargetBlock);
-  }
-  if (estimates_.contains(kHighPriorityTargetBlock)) {
-    return estimates_.at(kHighPriorityTargetBlock);
+    rate = estimates_.at(kMediumPriorityTargetBlock);
+  } else if (estimates_.contains(kHighPriorityTargetBlock)) {
+    rate = estimates_.at(kHighPriorityTargetBlock);
+  } else {
+    rate = IsTestnet() ? kFallbackTestnetFeeRate : kFallbackMainnetFeeRate;
   }
-  return IsTestnet() ? kFallbackTestnetFeeRate : kFallbackMainnetFeeRate;
+  return std::max(rate, kDustRelayFeeRate);
```

Comment:
`std::max(rate, kDustRelayFeeRate)` just lets us avoid fee rate eq to 1 for example, which would produce a tx based on low fee rate which would just never have a change to be accepted.

If we are fixing this let's always return 
`std::clamp(rate, reasonable min, reasonable max)` so we are safe from creating a tx with unreasonbly high fee and potential loss of funds